docs/MODDING.md

```diff
@@ -0,0 +1,191 @@
+
+## Making the changes
+
+The limit is your imagination. You can add new features, fix bugs, add new mods, or even change the way the game works. We will demonstrate this by creating a mod that changes the speed of the day/night cycle.
+
+Let's being by finding where the time is updated. Thankfully in the save editor we have a slider already hooked up to the time of day so we can check there for reference. The save editor file is at `soh/soh/Enhancements/debugger/debugSaveEditor.cpp`, if we do a quick search within that file for time we will find the following at line 400:
+
+```cpp
+const uint16_t dayTimeMin = 0;
+const uint16_t dayTimeMax = 0xFFFF;
+ImGui::SliderScalar("Time", ImGuiDataType_U16, &gSaveContext.dayTime, &dayTimeMin, &dayTimeMax);
+```
+
+So this tells us that `gSaveContext.dayTime` is what we're looking for. Let's now do a global search for this to see if we can find where it is updated. We find the following in `soh/src/code/z_kankyo.c` line 925:
+
+```cpp
+if (IS_DAY || gTimeIncrement >= 0x190) {
+    gSaveContext.dayTime += gTimeIncrement;
+} else {
+    gSaveContext.dayTime += gTimeIncrement * 2; // time moves twice as fast at night
+}
+```
+
+We can make a quick change to this code to verify this is indeed what we are looking for, lets multiply the the gTimeIncrement by 10:
+
+```diff
+if (IS_DAY || gTimeIncrement >= 0x190) {
+-    gSaveContext.dayTime += gTimeIncrement;
++    gSaveContext.dayTime += gTimeIncrement * 10;
+} else {
+-    gSaveContext.dayTime += gTimeIncrement * 2; // time moves twice as fast at night
++    gSaveContext.dayTime += gTimeIncrement * 2 * 10; // time moves twice as fast at night
+}
+```
+
+Rebuild the game and launch it, then load a save file. You should see that the time of day is now moving much faster. Terrific! While we could wrap this up and call it a day, we could make this user configurable by making a few more changes. I think a slider would be good for this, there's a slider in the cheat menu that we can use as a reference. Let's find it in `soh/soh/SohMenuBar.cpp` around line 1120:
+
+```cpp
+UIWidgets::EnhancementSliderFloat("Hookshot Reach Multiplier: %.1fx", "##gCheatHookshotReachMultiplier", "gCheatHookshotReachMultiplier", 1.0f, 5.0f, "", 1.0f, false);
+```
+
+The float values being passed in here are `minumum`, `maximum`, and `default` respectively. We'll make our minimum 0.2 to allow it to move slower, and our maximum 5.0 to allow it to move up to 5x faster. We'll also set the default to 1.0 so that it doesn't change the behavior by default. Copy this line and paste it below, then make the relevant changes:
```

Comment:
```suggestion
The float values being passed in here are `minimum`, `maximum`, and `default` respectively. We'll make our minimum 0.2 to allow it to move slower, and our maximum 5.0 to allow it to move up to 5x faster. We'll also set the default to 1.0 so that it doesn't change the behavior by default. Copy this line and paste it below, then make the relevant changes:
```